jwst/ramp_fitting/ramp_fit.py

```diff
@@ -242,26 +582,32 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
     #   is the number of given by the NFRAMES keyword, and is the number of
     #   frames averaged on-board for a group, i.e., it does not include the
     #   groupgap.
-    effintim, nframes, groupgap, dropframes1 = utils.get_efftim_ped(model)
+
+    effintim = (nframes + groupgap) * frame_time
 
     # Get GROUP DQ and ERR arrays from input file
-    gdq_cube = model.groupdq
+    gdq_cube = groupdq
     gdq_cube_shape = gdq_cube.shape
 
     # If all the pixels have their initial groups flagged as saturated, the DQ
     #   in the primary and integration-specific output products are updated,
     #   the other arrays in all output products are populated with zeros, and
     #   the output products are returned to ramp_fit(). If the initial group of
     #   a ramp is saturated, it is assumed that all groups are saturated.
-    first_gdq = model.groupdq[:,0,:,:]
+    first_gdq = groupdq[:,0,:,:]
     if np.all(np.bitwise_and( first_gdq, dqflags.group['SATURATED'] )):
-        new_model, int_model, opt_model = utils.do_all_sat( model, imshape,
-                                                            n_int, save_opt )
+        new_model, int_model, opt_model = utils.do_all_sat(inpixeldq, groupdq, imshape, n_int, save_opt )
+        if (save_opt):
+            actual_segments = 0
+            actual_CRs = 0
 
-        return new_model, int_model, opt_model
+        return new_model.data, new_model.dq, new_model.var_poisson, new_model.var_rnoise, new_model.err, \
+           int_model.data, int_model.dq, int_model.var_poisson, int_model.var_rnoise, int_model.err, int_model.int_times, \
+           opt_model.slope, opt_model.sigslope, opt_model.var_poisson, opt_model.var_rnoise, opt_model.yint, opt_model.sigyint, \
+           opt_model.pedestal, opt_model.weights, opt_model.crmag, actual_segments, actual_CRs
 
     # Get max number of segments fit in all integrations
-    max_seg = calc_num_seg(gdq_cube, n_int)
+    max_seg, num_CRs = calc_num_seg(gdq_cube, n_int)
```

Comment:
Is num_CRs used ?

Reply:
It is not used on this call but is used in the call from the OLS fit section.